src/gui/application.cpp

```diff
@@ -489,18 +489,15 @@ void Application::slotownCloudWizardDone(int res)
         _checkConnectionTimer.start();
         slotCheckConnection();
 
-        // If one account is configured: enable autostart
-#ifndef QT_DEBUG
```

Comment:
@elsiehupp Since you are removing this standard definition, would then `DEV_BUILD `be always on when I choose a Debug configuration in the IDE? I mean, I am not that experienced with CMake, but, I don't see any logic to enable this option for debug builds by default. And, I think, this is required, cause, otherwise, the autostart will be brought back if I forget to set this `DEV_BUILD `option to **ON**.

Reply:
I thought it would be more general to disable autostart on dev builds than just on debug builds. We could make it a or condition, so that autostart is disabled on dev builds and on debug builds.

Reply:
I agree with @allexzander . Sorry for the changes. Please also condition the auto start activation to not being a debug build.

Reply:
IIRC the way I initially wrote this pull request was with `DEV_SUFFIX` being triggered by certain values of `CMAKE_BUILD_TYPE` rather creating a new option with `DEV_BUILD`. FWIW it might be clearer to use the option name `USE_DEV_SUFFIX` rather than the option name `DEV_BUILD`.

That said, I could wrap the entire new `if()` statement in `#ifndef QT_DEBUG` again.

Reply:
No, the idea is that you want to activate the auto start only for release builds that are not built with the DEV_BUILD option active.

Reply:
@elsiehupp I'm still having problems with understanding when reading this. Could you keep everything as is, but, introduce a variable that you later check like it was before?

Reply:
@elsiehupp Also, in your current implementation `#ifndef QT_DEBUG` gives **false**. This will result in Release builds not get auto-started. This is not what we want. Probably, it's this define itself is bringing confusion. What you can do is to add a variable here:
```
#ifndef QT_DEBUG
    bool isReleaseBuild = true;
#else
    bool isReleaseBuild = false;
#endif
```

or

```
#ifdef QT_DEBUG
    bool isReleaseBuild = false;
#else
    bool isReleaseBuild = true;
#endif
```
and then, use this variable in the `if` right away.

Reply:
@allexzander I accidentally swapped the `true` and `false` values. FWIW you don’t actually need to create a variable. Preprocessing directives are extremely weird, but the following text:
```
if (
#ifdef QT_DEBUG
    true
#else
    false
#endif
    && someVariable) {
    doTheThing();
}
```
is converted to the equivalent of:
```
if (true && someVariable) {
    doTheThing();
}
```
or:
```
if (false && someVariable) {
    doTheThing();
}
```
at compile time depending on the macro value of `QT_DEBUG`. The main reason not to create a variable is that the variable is only used once in the compiled code, so there’s no reason to allocate any memory for it.

Granted, it would be tidier to do the above code as:
```
#ifdef QT_DEBUG
if (someVariable) {
    doTheThing();
}
#endif
```
I used the syntax that I did primarily because it was a minimal intervention in the code I had already written, and I didn’t feel like thinking too hard about boolean expressions. Having given it further thought, the following would probably be tidiest:
```
#ifndef QT_DEBUG
        // Enable autostart or not
        if (
            // If at least one account is configured:
            && accountMan->accounts().size() >= 1
            // Don't autostart development/test builds
            && !USE_DEV_SUFFIX
            // Don't auto start when not being 'installed'
            && (!Utility::isMac() || QCoreApplication::applicationDirPath().startsWith("/Applications/"))
            ) {
            Utility::setLaunchOnStartup(_theme->appName(), _theme->appNameGUI(), true);
        }
#endif
```
The weird thing being that `setLaunchOnStartup()` is only invoked for `true` and not for `false`, though this is an aspect of the preexisting code. If you actually do want to set `setLaunchOnStartup(…, false)`, the way I wrote it (with the weird inline directives) would probably be the tidiest way to allow an `else{}` statement.

I mean, alternately, `config.h.in` could include something like `#cmakedefine01 QT_DEBUG`, which should make `QT_DEBUG` equal `false` rather than something like `nullptr` (i.e. a compiler error) if it isn’t defined and consequently allow it to be used as a regular `const bool`. (I haven’t actually tested this, though.)